assets/app/scripts/services/data.js

```diff
@@ -299,21 +304,21 @@ angular.module('openshiftConsole')
   DataService.prototype.get = function(resource, name, context, opts) {
     resource = APIService.toResourceGroupVersion(resource);
     opts = opts || {};
-
+    var key = this._uniqueKeyForResourceContext(resource, name, context, _.get(opts, 'http.params'));
     var force = !!opts.force;
     delete opts.force;
 
     var deferred = $q.defer();
 
-    var existingData = this._data(resource, context);
+    var existingData = this._data(key);
 
     // If this is a cached resource (immutable resources only), ignore the force parameter
     if (this._isResourceCached(resource) && existingData && existingData.by('metadata.name')[name]) {
       $timeout(function() {
         deferred.resolve(existingData.by('metadata.name')[name]);
       }, 0);
     }
-    else if (!force && this._watchInFlight(resource, context) && this._resourceVersion(resource, context)) {
+    else if (!force && this._watchInFlight(key) && this._resourceVersion(key)) {
```

Comment:
I see this comment:
```JavaScript
// If this is a cached resource (immutable resources only), ignore the force parameter
if (this._isResourceCached(resource) && existingData && existingData.by('metadata.name')[name]) {
```
In my snippet above, the only item in the hash is `imagestreamimages`, so that is the only resource that ever returns `true`.  

Reading it this way implies that `imagestreamimages` would be the only thing we ever cache, but I think we mean the opposite.